File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_temperature_unit/DisplayTemperatureUnitFactoryTest.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_temperature_unit;

import static org.assertj.core.api.Assertions.assertThat;

import android.app.Application;

import org.junit.jupiter.api.Test;

import coolpharaoh.tee.speicher.tea.timer.core.settings.TemperatureUnit;

class DisplayTemperatureUnitFactoryTest {

    @Test
    void getDisplayTemperatureUnitFahrenheit() {
        final DisplayTemperatureUnitStrategy displayTemperatureUnitStrategy = DisplayTemperatureUnitFactory.get(TemperatureUnit.FAHRENHEIT, new Application());
        assertThat(displayTemperatureUnitStrategy).isInstanceOf(DisplayTemperatureUnitStrategyFahrenheit.class);
    }

    @Test
    void getDisplayTemperatureUnitCelsius() {
        final DisplayTemperatureUnitStrategy displayTemperatureUnitStrategy = DisplayTemperatureUnitFactory.get(TemperatureUnit.CELSIUS, new Application());
        assertThat(displayTemperatureUnitStrategy).isInstanceOf(DisplayTemperatureUnitStrategyCelsius.class);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_temperature_unit/DisplayTemperatureUnitFactoryTest.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_temperature_unit

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.core.settings.TemperatureUnit
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class DisplayTemperatureUnitFactoryTest {

    @Test
    fun getDisplayTemperatureUnitFahrenheit() {
        val displayTemperatureUnitStrategy = DisplayTemperatureUnitFactory[TemperatureUnit.FAHRENHEIT, Application()]
        assertThat(displayTemperatureUnitStrategy).isInstanceOf(DisplayTemperatureUnitStrategyFahrenheit::class.java)
    }

    @Test
    fun getDisplayTemperatureUnitCelsius() {
        val displayTemperatureUnitStrategy = DisplayTemperatureUnitFactory[TemperatureUnit.CELSIUS, Application()]
        assertThat(displayTemperatureUnitStrategy).isInstanceOf(DisplayTemperatureUnitStrategyCelsius::class.java)
    }
}